test/common/grpc/async_client_manager_impl_test.cc

```diff
@@ -21,6 +24,85 @@ namespace Envoy {
 namespace Grpc {
 namespace {
 
+class RawAsyncClientCacheTest : public testing::Test {
+public:
+  RawAsyncClientCacheTest()
+      : api_(Api::createApiForTest(time_system_)),
+        dispatcher_(api_->allocateDispatcher("test_thread")), client_cache_(*dispatcher_) {}
+
+  // advanceTimeAndRun moves the current time as requested, and then executes
+  // all executable timers in a non-deterministic order. This mimics real-time behavior in
+  // libevent if there is a long delay between libevent regaining control. Here we want to
+  // test behavior with a specific sequence of events, where each timer fires within a
+  // simulated second of what was programmed.
+  void waitForSeconds(int seconds) {
+    for (int i = 0; i < seconds; i++) {
+      time_system_.advanceTimeAndRun(std::chrono::seconds(1), *dispatcher_,
+                                     Event::Dispatcher::RunType::NonBlock);
+    }
```

Comment:
Why do this in a for loop vs just advanceTimeAndRun for the full time?

Reply:
Please take a look at the method comment, because advanceTimeAndRun won't wake up timer in the middle of the function. but a real time system will invoke timer as time advance.

Reply:
Ok thanks, I missed the comment